synapse/handlers/device.py

```diff
@@ -948,8 +948,16 @@ async def user_device_resync(
             devices = []
             ignore_devices = True
         else:
+            sid = await self.store.get_device_list_last_stream_id_for_remote(user_id)
```

Comment:
I originally assumed `sid` = session ID, let's be more specific.

```suggestion
            stream_id = await self.store.get_device_list_last_stream_id_for_remote(user_id)
```